reana/reana_dev/utils.py

```diff
@@ -704,37 +724,56 @@ def translate_pep440_to_semver2(pep440_version):
-
-        if version_files.get(HELM_VERSION_FILE):
-            next_version = next_version or bump_semver2_version(current_version)
-            files_to_update.append(version_files.get(HELM_VERSION_FILE))
-        elif version_files.get(PYTHON_VERSION_FILE):
-            next_version = next_version or bump_pep440_version(current_version)
-            files_to_update.append(version_files.get(PYTHON_VERSION_FILE))
-            if version_files.get(OPENAPI_VERSION_FILE):
-                files_to_update.append(version_files.get(OPENAPI_VERSION_FILE))
-        elif version_files.get(JAVASCRIPT_VERSION_FILE):
-            next_version = next_version or bump_semver2_version(current_version)
-            files_to_update.append(version_files.get(JAVASCRIPT_VERSION_FILE))
-
-        for file_ in files_to_update:
-            replace_string(
-                file_=file_,
-                find=current_version,
-                replace=next_version,
-                component=component,
+def bump_component_version(
+    component: str, next_version: str = None
+) -> (str, List[str]):
+    """Bump to next component version.
+
+    If next_version is set, it will be used for the component.
+    If not, version will be bump automatically from the current one.
+
+    :param component: component to bump
+    :param next_version: new PEP440 version
+    :type component: str
+    :type next_version: str
+
+    :return: next_version in SemVer2 format and list of modified version files
+    :rtype: str, List[str]
+    """
+    if next_version and not parse_pep440_version(next_version):
+        raise Exception("Version is not in PEP440 format")
+
+    version_files = get_component_version_files(component)
+    updated_files = []
+
+    for file_type, file_path in version_files.items():
+        if not file_path:
+            continue
+
+        current_version = get_current_component_version_from_source_files(
+            component, version_file=file_type
+        )
+
+        if file_type in [HELM_VERSION_FILE, JAVASCRIPT_VERSION_FILE]:
+            new_version = (
+                translate_pep440_to_semver2(next_version)
+                if next_version
+                else bump_semver2_version(current_version)
             )
+        elif file_type in [PYTHON_VERSION_FILE, OPENAPI_VERSION_FILE]:
+            new_version = next_version or bump_pep440_version(current_version)
+        else:
+            raise Exception(f"Cannot bump the following {file_type} file")
 
-        return next_version, files_to_update
-    except Exception as e:
-        display_message(
-            f"Something went wront while bumping the version: {e}", component
+        replace_string(
+            file_=file_path,
+            find=current_version,
+            replace=new_version,
+            component=component,
         )
+        updated_files.append(file_path)
+
+    return translate_pep440_to_semver2(next_version) if next_version else next_version, updated_files
```

Comment:
When I did:
``` 
cd reana
reana-dev git-create-release-commit -c .
```
It executed `git commit -m 'release: None'` because the `next_version` was `None`.
I guess we should return `new_version` in case `next_version` is not present:
Perhaps `next_version = next_version or new_version` before the return statement ?

Also, what is the reason behind `translate_pep440_to_semver2(next_version)` ? If we are specifying `pep440` version it will return as semver2?

Reply:
Will fix it. 

Regarding the last question, REANA is using SemVer2. And to allow to use of `next_version` with Python (PEP440) I wanted to enforce usage of the PEP440 format because, in the end, it can be translated to SemVer2.

Reply:
The returned version of this function will be only reflected in the release commit message. Different components currently uses different types of versions for releases. For example do `git log --all --grep='release'` inside `reana` and `reana-server`, you will see that `reana` uses SemVer2 while `reana-server` pep440. So this change enforces to use SemVer2 in all the components. Perhaps it's better to remove the translate function here at all?

Reply:
I see. In case I will remove the translate function, the Python update will not work if SemVer2 is specified as `next_version` because we do not have a `semver2 => pep440` translate function.

Reply:
In addition, the differences in git tags for `release` are because of the bug (or not defined behavior) in `bump_component_version`. As mentioned by @tiborsimko in the https://github.com/reanahub/reana/issues/516 , git tags follow SemVer2 and, from what you posted above, not every `release` follows this (some tags are PEP440, some SemVer2)